analyzers/src/SonarAnalyzer.CSharp/Helpers/CSharpNavigationHelper.cs

```diff
@@ -77,7 +77,26 @@ public static IEnumerable<SwitchSectionSyntax> GetPrecedingSections(this SwitchS
 
         #endregion Switch
 
-        public static StatementSyntax GetPrecedingStatement(this StatementSyntax currentStatement) =>
-            currentStatement.Parent.ChildNodes().OfType<StatementSyntax>().TakeWhile(x => x != currentStatement).LastOrDefault();
+        public static StatementSyntax GetPrecedingStatement(this StatementSyntax currentStatement)
+        {
+            var previousStatement = currentStatement.Parent
+                                                    .ChildNodes()
+                                                    .OfType<StatementSyntax>()
+                                                    .TakeWhile(x => x != currentStatement)
+                                                    .LastOrDefault();
+            if (previousStatement == null) // this means that we might be in a top-level-statement
```

Comment:
Yes, that's right. The function detects the previous statement that is at the same level.
I think it might be better if I move it into the only rule that it's actually using it and rename it since it's not really generic in its usage.
The use of `StatementSyntax` in the input is only because this function is used for `IfStatement` and `SwitchStament` . WDYT?

Reply:
In the in the other rule you changed, you had a different way to check if the `node` is on the top level or not. Can't we use that also here? I think that having to maintain multiple ways to check this will prove hard and confusing.